src/toil/jobStores/aws/jobStore.py

```diff
@@ -681,6 +688,21 @@ def bucket_creation_pending(e):
                             raise
                         else:
                             return None
+                    elif e.status == 301:
+                        # This is raised if the user attempts to get a bucket in a region outside
+                        # the specified one, if the specified one is not `us-east-1`.  The us-east-1
+                        # server allows a user to use buckets from any region.
+                        if self.region == 'us-east-1':
+                            message = ('Region is us-east-1 but 301 error was raised when '
+                                       'attempting to create the bucket. 301 Errors are usually '
+                                       'raised when a non-us-east-1 connection attempts to get a '
+                                       'bucket from another region. Cannot proceed further.')
+                        else:
+                            b = self.s3.get_bucket(bucket_name, validate=False)
+                            message = ('A bucket with the same name as the jobstore was found in '
+                                       'another region (%s). Cannot proceed further as the unique '
+                                       'jobstore name is already claimed.' % b.get_location())
```

Comment:
I think you can only call get_location() if the connection is to us-east-1. if I am right, then this branch of the control flow is not covered by tests.